### ISSUE TITLE
Exponential grad norm growth as sequence length increase

### ISSUE THREAD

**Issue by victkk** [CUT3R/CUT3R#88]
Thanks in advance!
I'm currently finetuning your model(added another DPT head for other per point attribute prediction and some new loss function). Training with 4 views works fine(The grad norm is about 1 ). But as training view increases to 16, grad norm suddenly explode to the magnitude of 100000. Since the loss is only slightly bigger in the 16 views training, I think this has something to do with the RNN nature of CUT3R.

Have you met this problem in your training? And do you have any suggestions on how to solve this?

**Comment by Forrest-110** [comment]
Hi @victkk! It sounds like it's running into gradient explosion due to the RNN’s recurrent nature. If you don’t want to change the architecture, gradient norm clipping seems to be the only fix. If you’re open to modifying the model, I’d suggest adding a forget gate and turning the RNN into a GRU or LSTM.
